user-management/src/test/java/de/tum/in/www1/artemis/usermanagement/security/jwt/MockTokenProvider.java

```diff
@@ -0,0 +1,67 @@
+package de.tum.in.www1.artemis.usermanagement.security.jwt;
+
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.util.Date;
+import java.util.stream.Collectors;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.util.ObjectUtils;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import tech.jhipster.config.JHipsterProperties;
+
+/**
+ * This class implements token creation in order to test the token verification,
+ * since it is not included in {@link TokenProvider} because the Gateway is not responsible for authenticating users.
+ */
+public class MockTokenProvider {
```

Comment:
This doesn't look like a mock but rather like a full implementation. Could this be simplified for testing?
=> If the actual implementation would change, this would have to change as well...

Reply:
Additionally, is this a C&P error? Since we are not in the Gateway but the use management?

Reply:
Yes, copy and paste error. I think that in order to test it properly we need full implementation

Reply:
But then maybe don't call it a "mock". Maybe "TokenGenerator" or "TestTokenGenerator"?